src/main/scala/dotty/dokka/ScalaContentBuilder.scala

```diff
@@ -0,0 +1,504 @@
+
+        def table(
+            kind: Kind = ContentKind.Main,
+            sourceSets: Set[DokkaConfiguration$DokkaSourceSet] = mainSourcesetData,
+            styles: Set[Style] = mainStyles,
+            extra: PropertyContainer[ContentNode] = mainExtra,
+            headers: List[ContentGroup] = defaultHeaders
+            )(
+            buildBlock: Function1[ScalaPageContentBuilder#ScalaTableBuilder, ScalaPageContentBuilder#ScalaTableBuilder]
+            ) = addChild(
+            ContentTable(
+                headers.asJava,
+                buildBlock(ScalaTableBuilder(mainDRI, sourceSets, kind, styles, extra)).build().asJava,
+                DCI(mainDRI.asJava, kind),
+                sourceSets.asJava,
+                styles.asJava,
+                extra
+            )
+        )
+
+        def text(
+            text: String,
+            kind: Kind = ContentKind.Main,
+            sourceSets: Set[DokkaConfiguration$DokkaSourceSet] = mainSourcesetData,
+            styles: Set[Style] = mainStyles,
+            extra: PropertyContainer[ContentNode] = mainExtra
+        ) = addChild(
+            buildText(text, kind, sourceSets, styles, extra)
+        )
+
+        private def buildText(
+            text: String,
+            kind: Kind = ContentKind.Main,
+            sourceSets: Set[DokkaConfiguration$DokkaSourceSet] = mainSourcesetData,
+            styles: Set[Style] = mainStyles,
+            extra: PropertyContainer[ContentNode] = mainExtra
+        ) = ContentText(text, DCI(mainDRI.asJava, kind), sourceSets.asJava, styles.asJava, extra)
+
+        def groupingBlock[A, T <: Documentable, G <: List[(A, List[T])]](
+            name: String,
+            elements: G,
+            kind: Kind = ContentKind.Main,
+            sourceSets: Set[DokkaConfiguration$DokkaSourceSet] = mainSourcesetData,
+            styles: Set[Style] = Set(),
+            extra: PropertyContainer[ContentNode] = mainExtra,
+            renderWhenEmpty: Boolean = false,
+            needsSorting: Boolean = true,
+            headers: List[ContentGroup] = List(),
+            needsAnchors: Boolean = true,
+            omitSplitterOnSingletons: Boolean = true
+        )(
+            groupSplitterFunc: (ScalaPageContentBuilder#ScalaDocumentableContentBuilder, A) => ScalaPageContentBuilder#ScalaDocumentableContentBuilder
+        )(
+            elementFunc: (ScalaPageContentBuilder#ScalaDocumentableContentBuilder, T) => ScalaPageContentBuilder#ScalaDocumentableContentBuilder
+        ) = if (renderWhenEmpty || !elements.isEmpty) {
+            header(3, name, kind, styles = styles, extra = extra plus SimpleAttr.Companion.header(name))()
+            .group(styles = Set(ContentStyle.WithExtraAttributes), extra = extra plus SimpleAttr.Companion.header(name)){ bdr =>
+                elements.foldLeft(bdr){ (b, groupped) =>
+                    val (key, values) = groupped
+                    (if(values.size > 1) groupSplitterFunc(b, key) else b)
```

Comment:
Note: with Scala 3, conditionals can be written:

```scala
if condition then whenTrue() else whenFalse()
```

I think most of the codebase is actually using this format rather than `if (cond) whenTrue() else whenFalse()`. It'd be nice if we could keep it consistent.